utils/analyticsmetrics.go

```diff
@@ -0,0 +1,204 @@
+		return false
+	}
+	// Verify xsc version.
+	xscVersion, err := ams.xscManager.GetVersion()
+	if err != nil {
+		return false
+	}
+	if err = clientutils.ValidateMinimumVersion(clientutils.Xsc, xscVersion, xscservices.AnalyticsMetricsMinXscVersion); err != nil {
+		return false
+	}
+	return true
+}
+
+func (ams *AnalyticsMetricsService) XscManager() *xsc.XscServicesManager {
+	return ams.xscManager
+}
+
+func (ams *AnalyticsMetricsService) SetMsi(msi string) {
+	ams.msi = msi
+}
+
+func (ams *AnalyticsMetricsService) GetMsi() string {
+	return ams.msi
+}
+
+func (ams *AnalyticsMetricsService) SetStartTime() {
+	ams.startTime = time.Now()
+}
+
+func (ams *AnalyticsMetricsService) GetStartTime() time.Time {
+	return ams.startTime
+}
+
+func (ams *AnalyticsMetricsService) ShouldReportEvents() bool {
+	return ams.shouldReportEvents
+}
+
+func (ams *AnalyticsMetricsService) FinalizeEvent() *xscservices.XscAnalyticsGeneralEventFinalize {
+	return ams.finalizeEvent
+}
+
+func (ams *AnalyticsMetricsService) SetFinalizeEvent(finalizeEvent *xscservices.XscAnalyticsGeneralEventFinalize) {
+	ams.finalizeEvent = finalizeEvent
+}
+
+func (ams *AnalyticsMetricsService) CreateGeneralEvent(product xscservices.ProductName, eventType xscservices.EventType) *xscservices.XscAnalyticsGeneralEvent {
+	osAndArc, err := coreutils.GetOSAndArc()
+	curOs, curArch := "", ""
+	if err != nil {
+		log.Debug(fmt.Errorf("failed to get os and arcitucture for general event request to XSC service, error: %s ", err.Error()))
+	} else {
+		splitOsAndArch := strings.Split(osAndArc, "-")
+		curOs = splitOsAndArch[0]
+		curArch = splitOsAndArch[1]
+	}
+
+	event := xscservices.XscAnalyticsGeneralEvent{
+		XscAnalyticsBasicGeneralEvent: xscservices.XscAnalyticsBasicGeneralEvent{
+			EventType:              eventType,
+			EventStatus:            xscservices.Started,
+			Product:                product,
+			JfrogUser:              ams.xscManager.Config().GetServiceDetails().GetUser(),
+			OsPlatform:             curOs,
+			OsArchitecture:         curArch,
+			AnalyzerManagerVersion: GetAnalyzerManagerVersion(),
+		},
+	}
+	return &event
+}
+
+func (ams *AnalyticsMetricsService) AddGeneralEvent(event *xscservices.XscAnalyticsGeneralEvent) {
+	if !ams.ShouldReportEvents() {
+		log.Debug("A general event request was not sent to XSC - analytics metrics are disabled.")
+		return
+	}
+	msi, err := ams.xscManager.AddAnalyticsGeneralEvent(*event)
+	if err != nil {
+		log.Debug(fmt.Errorf("failed sending general event request to XSC service, error: %s ", err.Error()))
+		return
+	}
+	log.Debug(fmt.Sprintf("New General event added successfully. multi_scan_id %s", msi))
+	// Set event's analytics data.
+	ams.SetMsi(msi)
+	ams.SetStartTime()
+}
+
+func (ams *AnalyticsMetricsService) UpdateGeneralEvent(event *xscservices.XscAnalyticsGeneralEventFinalize) {
+	if !ams.ShouldReportEvents() {
+		log.Debug("A general event update request was not sent to XSC - analytics metrics are disabled.")
+		return
+	}
```

Comment:
```suggestion
	if !ams.ShouldReportEvents() {
		return
	}
```
No need to log at the end, we already logged that it is disabled

Reply:
I think that its ok to add in debug mode also if we are skipping the updating.